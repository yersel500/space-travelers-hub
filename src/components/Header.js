// import { useDispatch } from 'react-redux';
import {
  NavLink,
} from 'react-router-dom';
import logo from '../assets/planet.png';
// import { FetchData } from '../redux/missions/MissionReducer';

const Header = () => {
  const activeStyle = {
    textDecoration: 'underline',
  };
  // const dispatch = useDispatch();
  // const loadMission = () => {
  //   dispatch(FetchData());
  // };

  return (
    <header>
      <nav>
        <div className="title-container">
          <img className="logo" src={logo} alt="logo" />
          <h1 className="header-title">Space Travelers&apos; Hub</h1>
        </div>
        <ul className="list-container">
          <li>
            <NavLink
              to="/"
              className="rockets-nav"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink
              to="missions"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
              // onClick={loadMission}
            >
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink
              to="myprofile"
              className="my-profile-nav"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </nav>
      <hr />
    </header>
  );
};

export default Header;
