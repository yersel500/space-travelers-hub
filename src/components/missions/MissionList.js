// import { useEffect } from 'react';
import { useSelector } from 'react-redux';
// import { FetchData } from '../../redux/missions/MissionReducer';
import MissionItem from './MissionItem';

const MissionList = () => {
  // const dispatch = useDispatch();
  const myState = useSelector((state) => state.missionReducer);
  // useEffect(() => {
  //   if (myState.lenght === 0) {
  //     dispatch(FetchData());
  //   }
  // }, []);

  return (
    <table>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>Join</th>
        </tr>
      </thead>
      <tbody>
        {myState.map((element) => (
          <MissionItem
            key={element.MissionId}
            id={element.MissionId}
            mission={element.MissionName}
            description={element.description}
            reserved={element.reserved}
          />
        ))}
      </tbody>
    </table>
  );
};

export default MissionList;
