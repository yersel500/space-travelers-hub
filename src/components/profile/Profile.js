import { useSelector } from 'react-redux';

const Profile = () => {
  const rockets = useSelector((state) => state.rocketsReducer);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <div className="profile-container">
      <div className="rockets-profile-container">
        <h2 className="rockets-profile-title">My Rockets</h2>
        <ul className="rockets-ul">
          {reservedRockets.map((rocket) => <li className="rockets-li" key={rocket.id}>{rocket.name}</li>)}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
