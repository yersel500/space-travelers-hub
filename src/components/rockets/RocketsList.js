import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getRockets, rocketsData } from '../../redux/rockets/RocketsReducer';
import Rocket from './Rocket';

const Rockets = () => {
  const rocketStore = useSelector((state) => state.rocketsReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    async function fetchData() {
      if (rocketStore.length === 0) {
        const api = await getRockets();
        dispatch(rocketsData(api));
      }
    }
    fetchData();
  }, []);

  return (
    <div>
      {rocketStore.map((rocket) => (
        <Rocket
          name={rocket.name}
          description={rocket.description}
          key={rocket.id}
          id={rocket.id}
          image={rocket.image}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
};

const RocketsList = () => (
  <div>
    <Rockets />
  </div>
);

export default RocketsList;
