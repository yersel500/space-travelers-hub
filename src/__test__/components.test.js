import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';
import RocketsList from '../components/rockets/RocketsList';
import store from '../redux/configureStore';
import MissionList from '../components/missions/MissionList';
import Profile from '../components/profile/Profile';

describe('Testing Components', () => {
  test('Render Header', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Rockets section', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketsList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Missions section', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MissionList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Render Profile section', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
