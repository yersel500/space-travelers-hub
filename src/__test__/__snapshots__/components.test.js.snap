// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components Render Header 1`] = `
<header>
  <nav>
    <div
      className="title-container"
    >
      <img
        alt="logo"
        className="logo"
        src="planet.png"
      />
      <h1
        className="header-title"
      >
        Space Travelers' Hub
      </h1>
    </div>
    <ul
      className="list-container"
    >
      <li>
        <a
          aria-current="page"
          className="rockets-nav active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="my-profile-nav"
          href="/myprofile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
  <hr />
</header>
`;

exports[`Testing Components Render Missions section 1`] = `
<table
  className="missions-container"
>
  <thead>
    <tr>
      <th
        style={
          Object {
            "width": "10%",
          }
        }
      >
        Mission
      </th>
      <th
        style={
          Object {
            "width": "60%",
          }
        }
      >
        Description
      </th>
      <th
        style={
          Object {
            "width": "15%",
          }
        }
      >
        Status
      </th>
      <th>
        Join
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Testing Components Render Profile section 1`] = `
<div
  className="profile-container"
>
  <div
    className="rockets-profile-container"
  >
    <h2
      className="rockets-profile-title"
    >
      My Missions
    </h2>
    <div
      className="rockets-ul"
    />
  </div>
  <div
    className="rockets-profile-container"
  >
    <h2
      className="rockets-profile-title"
    >
      My Rockets
    </h2>
    <ul
      className="rockets-ul"
    />
  </div>
</div>
`;

exports[`Testing Components Render Rockets section 1`] = `
<div>
  <div />
</div>
`;
